lib/electr/repl/reader.rb

```diff
@@ -9,6 +9,7 @@ def initialize(readline_lib = Readline)
 
     def run
       line = @readline.readline(@prompt, true)
+      return if line.nil?
```

Comment:
Hey, mate,

Imhi, exiting ( and saying bye ) should be achieved outside of the repl,
because it already has the responsibility of coordinating the reader and
printer.

Thinking of it, using a throw ( catched above to display a bye message )
instead of returning true in run once could be a cleaner way to do it ( I'm
not happy with mixing pn and true/false in run once results).

Let me know your opinion 😁


Le dim. 8 nov. 2015 20:36, Xavier Nayrac <notifications@githu
<notifications@github.com>

b.com <notifications@github.com>> a écrit :

In lib/electr/repl/reader.rb
<https://github.com/lkdjiin/electr/pull/19#discussion_r44229131>:

> @@ -9,6 +9,7 @@ def initialize(readline_lib = Readline) > > def run >
line = @readline.readline(@prompt, true) > + return if line.nil?

This trick is the only one needed, isn't it? There is no need to modify the
code in repl.rb. Because readline returns nil on an end of file one can
surely do:

line or exit

and that's it! Actually that cannot be a direct call to exit because we
have to say «Bye» like for ctrl-c but this is the idea. What do you think?

—
Reply to this email directly or view it on GitHub
<https://github.com/lkdjiin/electr/pull/19/files#r44229131>.<img src="
https://ci6.googleusercontent.com/proxy/uY1x4S8Gb-lCru_Tl7v02R6IWJZVRbheWbCRzgnnWtdxxDWElVEb4X-rUhpsH2D0gmIksXQ6lvK9Ay8ft2L5hakcUtj-ccnmF-YgChN13UV5pwP0XSUwEVGDvh4AkWvUXFJOI4lAFND5BT3KQVAcAft9DBk5qw=s0-d-e1-ft#https://github.com/notifications/beacon/AAWVr33-U-y6WaY4S6x60xkgxNjlrMfOks5pD5uzgaJpZM4GdxaI.gif
">